packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -73,18 +75,52 @@ const ruleset: any = {
         function: longRunningResponseStatusCodeArm,
       },
     },
+
     // RPC Code: RPC-Async-V1-02
-    ProvisioningStateSpecified: {
-      description: 'A LRO PUT and PATCH operations response schema must have "ProvisioningState" property specified.',
+    //PUT
+    ProvisioningStateSpecifiedForLROPut: {
+      description:
+        'A LRO PUT operation\'s response schema must have "ProvisioningState" property specified for the 200 and 201 status codes.',
+      message: "{{error}}",
+      severity: "error",
+      resolved: true,
+      formats: [oas2],
+      given: ["$[paths,'x-ms-paths'].*[put][?(@property === 'x-ms-long-running-operation' && @ === true)]^"],
+      then: {
+        function: provisioningStateSpecifiedForLROPut,
+      },
+    },
+    //Patch
+    ProvisioningStateSpecifiedForLROPatch: {
+      description:
+        'A long running Patch operation\'s response schema must have "ProvisioningState" property specified for the 200 status code.',
       message: "{{error}}",
       severity: "error",
       resolved: true,
       formats: [oas2],
-      given: "$[paths,'x-ms-paths'].*[put,patch].[?(@property === 'x-ms-long-running-operation' && @ === true)]^^",
+      given: [
+        "$[paths,'x-ms-paths'].*[patch][?(@property === 'x-ms-long-running-operation' && @ === true)]^.responses[?(@property == '200')]",
+      ],
+      then: {
+        function: provisioningStateSpecifiedForLROPatch,
+      },
+    },
+    //Delete
+    ProvisioningStateSpecifiedForLRODelete: {
+      description:
+        'A long running Delete operation\'s response schema must have "ProvisioningState" property specified for the 200 status code.',
+      message: "{{error}}",
+      severity: "warn",
+      resolved: true,
+      formats: [oas2],
+      given: [
+        "$[paths,'x-ms-paths'].*[delete][?(@property === 'x-ms-long-running-operation' && @ === true)]^.responses[?(@property == '200')]",
+      ],
       then: {
-        function: provisioningStateSpecified,
+        function: provisioningStateSpecifiedForLRODelete,
```

Comment:
Instead of using a custom function here, you can use the core function [`truthy`](https://meta.stoplight.io/docs/spectral/cb95cf0d26b83-core-functions#truthy).
```suggestion
        field: "provisioningState",
        function: truthy,
```